software/bootcamp-code/main.py

```diff
@@ -33,56 +56,170 @@ def main_loop():
-                # Find the coordinates of the largest ball.
-                # (THE ZERO COORDINATES OF THE FRAME ARE IN THE UPPER LEFT CORNER.)
-                # Using these coordinates calculate the side speed, forward speed and rotation for the robot.
+                    
+                # The biggest ball is the last one in the detected balls list.
+                # Using the coordinates of the biggest ball calculate the side speed, forward speed and rotation for the robot.
                 
                 # The destination coordinates are the difference between the ball location and desired location.
-                dest_x = ball_desired_x - processedData.balls[0].x
-                dest_y = ball_desired_y - processedData.balls[0].y
-                # Normalize destination in the [-1;1] range and and then multiply it with motor max speed.
-                # (then the driving is proportional)
-                # Speed range for motors is 48 - 2047, we use 100 for max motor speed at the moment.
-                x_speed = (dest_x/cam.rgb_width) * 100
-                y_speed = (dest_y/cam.rgb_height) * 100
-                #rotation = 0
-                # Drive towards a ball:
-                    # side speed is x_speed
-                    # forward speed is y_speed
-                    # rotation if you want to turn
-                #print("X: ", x_speed, "Y: ", y_speed)
-                omni_motion.move(x_speed, y_speed, 0)
+                dest_x = ball_desired_x - processedData.balls[-1].x
+                dest_y = ball_desired_y - processedData.balls[-1].y
+              
+                # if ball is close enough, circle it and find a basket (the distance is from the 0 coordinate - closer is larger value)
+                if processedData.balls[-1].distance>400:
+                    state=State.FIND_BASKET
+                    continue
+                # else simultaniously drive to and center the ball
+                else:
+                    # Normalize destination in the [-1;1] range and and then multiply it with motor max speed.
+                    # (then the driving is proportional)
+                    x_speed = (dest_x/cam.rgb_width) * max_motor_speed
+                    y_speed = (dest_y/cam.rgb_height) * max_motor_speed
+                    omni_motion.move(-1*x_speed, -1*y_speed, x_speed)
+            
+            # state to find the basket when ball is found - circle around the ball until basket is found, then move on to throwing
+            elif state==State.FIND_BASKET:
+                last_state, new_state =  state_printer(state, last_state, new_state)
+                # if there are no balls, then find one
+                if len(processedData.balls)<1:
+                    state=State.FIND_BALL
+                    continue
+                    
+                # find blue basket
+                if basketColor==basketColor.BLUE:
```

Comment:
How should the basket color be swapped? Maybe you can switch the target basked based on the current target color